codegen/cmd/injection-gen/generators/filtered_informer.go

```diff
@@ -246,7 +248,7 @@ func (w *wrapper) List(selector {{ .labelsSelector|raw }}) (ret []*{{ .type|raw
 		return nil, err
 	}
 	for idx := range lo.Items {
-		ret = append(ret, &lo.Items[idx])
+		ret = append(ret, {{if not .listerHasPointerElem}}&{{end}}lo.Items[idx])
```

Comment:
what happens to the append if the `not` condition is not true

Reply:
Depending on the condition it will be either

`&lo.Items[idx]` 
or
`lo.Items[idx]` 

We had to do this to support code generation for istio types

Reply:
sounds good